decompiler/pipeline/preprocessing/remove_stack_canary.py

```diff
@@ -46,7 +46,18 @@ def _patch_canary(self, node: BasicBlock):
         Patch Branches to stack fail node.
         """
         for pred in self._cfg.get_predecessors(node):
-            self._patch_branch_condition(pred)
+            self._remove_empty_block_between(pred)
+        self._cfg.remove_node(node)
+
+    def _remove_empty_block_between(self, node: BasicBlock) -> None:
+        """
+        Removes empty nodes between stack fail and branch recursively.
+        """
+        if not node.is_empty():
+            self._patch_branch_condition(node)
+            return
+        for pred in self._cfg.get_predecessors(node):
+            self._remove_empty_block_between(pred)
```

Comment:
Looks good so far!

Still, tests are missing.
Could you please test with 2 cases:
- one empty block in between
- two (or three) - more than one - empty blocks in between.

Reply:
Added a few tests (the one you wanted + one with a wrong cfg + one with multiple paths to __stack_chk_fail.